superset/models/sql_lab.py

```diff
@@ -203,6 +203,10 @@ def sqlalchemy_uri(self) -> URL:
     def url(self) -> str:
         return "/superset/sqllab?savedQueryId={0}".format(self.id)
 
+    @property
+    def sql_tables(self) -> List[Table]:
+        return list(ParsedQuery(self.sql).tables)
```

Comment:
It's too bad there isn't a great pattern to memoise this on the object that I'm aware of. Parsing queries gets expensive and I'd hate to see this called multiple times for the same object in a web request.

Reply:
Seems premature, and unclear if calling a caching backend would perform better than doing the actual parsing...

Reply:
I was referring to the equivalent of Ruby's:

Reply:
This isn't a blocker.

Reply:
Added an lru cache, not a big win since we will only cache the parsed query on an object instance itself. But will avoid possible repeated property get's during the instance lifetime.

Reply:
Actually reverted this because of `mypy`
https://github.com/python/mypy/issues/1362

Reply:
That's fine, like I said it's not a blocker :)